e2e/nomostest/prometheus_metrics.go

```diff
@@ -0,0 +1,653 @@
+		prometheusmodel.LabelName(ocmetrics.KeyStatus.Name()):          prometheusmodel.LabelValue(status),
+		// TODO: Add "commit"?
+	}
+	query := fmt.Sprintf("%s%s", metricName, labels)
+	return metricExists(ctx, nt, v1api, query)
+}
+
+func metricApplyOperationsViewHasValueAtLeast(ctx context.Context, nt *NT, v1api prometheusv1.API, reconcilerPodName, operation, kind, status string, value int) error {
+	metricName := ocmetrics.ApplyOperationsView.Name
+	metricName = fmt.Sprintf("%s%s", prometheusConfigSyncMetricPrefix, metricName)
+	labels := prometheusmodel.LabelSet{
+		prometheusmodel.LabelName(ocmetrics.KeyComponent.Name()):       prometheusmodel.LabelValue(ocmetrics.OtelCollectorName),
+		prometheusmodel.LabelName(ocmetrics.ResourceKeyPodName.Name()): prometheusmodel.LabelValue(reconcilerPodName),
+		prometheusmodel.LabelName(ocmetrics.KeyController.Name()):      prometheusmodel.LabelValue(ocmetrics.ApplierController), // TODO: validate "remeditor" operations
+		prometheusmodel.LabelName(ocmetrics.KeyOperation.Name()):       prometheusmodel.LabelValue(operation),
+		prometheusmodel.LabelName(ocmetrics.KeyType.Name()):            prometheusmodel.LabelValue(kind),
+		prometheusmodel.LabelName(ocmetrics.KeyStatus.Name()):          prometheusmodel.LabelValue(status),
+		// TODO: Add "commit"?
+	}
+	// ApplyOperationsView is a count, so we don't need to aggregate
+	query := fmt.Sprintf("%s%s", metricName, labels)
+	return metricExistsWithValueAtLeast(ctx, nt, v1api, query, float64(value))
+}
+
+func metricRemediateDurationViewHasStatus(ctx context.Context, nt *NT, v1api prometheusv1.API, reconcilerPodName, kind, status string) error {
+	metricName := ocmetrics.RemediateDurationView.Name
+	// RemediateDurationView is a distribution. Query count to aggregate.
+	metricName = fmt.Sprintf("%s%s%s", prometheusConfigSyncMetricPrefix, metricName, prometheusDistributionCountSuffix)
+	labels := prometheusmodel.LabelSet{
+		prometheusmodel.LabelName(ocmetrics.KeyComponent.Name()):       prometheusmodel.LabelValue(ocmetrics.OtelCollectorName),
+		prometheusmodel.LabelName(ocmetrics.ResourceKeyPodName.Name()): prometheusmodel.LabelValue(reconcilerPodName),
+		prometheusmodel.LabelName(ocmetrics.KeyType.Name()):            prometheusmodel.LabelValue(kind),
+		prometheusmodel.LabelName(ocmetrics.KeyStatus.Name()):          prometheusmodel.LabelValue(status),
+		// TODO: Add "commit"?
+	}
+	query := fmt.Sprintf("%s%s", metricName, labels)
+	return metricExists(ctx, nt, v1api, query)
+}
+
+func metricExists(ctx context.Context, nt *NT, v1api prometheusv1.API, query string) error {
+	ctx, cancel := context.WithTimeout(ctx, prometheusQueryTimeout)
+	defer cancel()
+
+	nt.DebugLogf("prometheus query: %s", query)
+	response, warnings, err := v1api.Query(ctx, query, time.Now())
+	if err != nil {
+		return err
+	}
+	if len(warnings) > 0 {
+		nt.T.Logf("prometheus warnings: %v", warnings)
+	}
```

Comment:
Move these lines into a seperate function since they are used multiple times.

Reply:
Extracted metricQueryNow & metricResultMustExist funcs. It may help consistency, but I'm not sure it really helps readability.